common/src/sql/dbinit.sql

```diff
@@ -39,6 +39,14 @@ CREATE DATABASE omicron;
 CREATE USER omicron;
 ALTER DEFAULT PRIVILEGES GRANT INSERT, SELECT, UPDATE, DELETE ON TABLES to omicron;
 
+-- Fleet: Represents a collection of racks
```

Comment:
It seemed like one of those "seems nice to have now, is easy to drop if we don't need it, but would be difficult to add if we didn't have it" sorta things. It kinda reminded me of when I was adding references to an implicit "rack ID", but you suggested making it an explicit object in the database -- although we'll certainly be operating on *one* rack in the forseeable future, and *one* fleet of racks beyond that, having a way to distinguish fleet ownership seemed like a handy concept to have.

We definitely are still hard-coding `FLEET_ID`, but those spots are easy to find, because they're all still using the constant. I figured that we could eventually phase-out the constant, and generate a fleet ID dynamically during rack setup (or find an existing fleet ID, if adding a rack to an existing fleet). At least, I thought this would let us identify when two racks should or should not be considering themselves part of the same "group".

If you'd prefer, an alternative could be:
- Have a `resource_usage` object which exists with the hard-coded `FLEET_ID`, even with no corresponding "fleet" object.
- Delete all references to "fleet ID" as a foreign key in the rack/silo tables.
- Figure out some other entity to identify if "two racks are part of the same fleet, or part of a different fleet" for e.g. initialization.

Reply:
> Have a resource_usage object which exists with the hard-coded FLEET_ID, even with no corresponding "fleet" object.

I believe this is what we do with fleet-level role assignments, FWIW.  We use the hardcoded fleet_id but there's no record anywhere that has that id.  It's a little strange.  But I think it's equally strange to have a table that is constrained to contain exactly one row. 🤷 

If there's a concrete reason we expect the database to contain information about stuff from another Fleet, then it makes sense to have a Fleet table here with fleet_id foreign keys, etc.  If we don't, I feel like it makes sense to drop the table and the foreign keys because it feels like just another piece of data that could become wrong.  Like, what if we decided there should be only one Fleet in the database, but we kept the `fleet_id` foreign key, and then discovered a `fleet_id` with some unexpected fleet id?  Or two records in the Fleet table?

It's a good question whether it's easier to add now vs. later.  If we do this later, it feels like we could define a schema migration that adds a column to the "rack" table and populates the fleet_id value for all existing rows.  My fear is that if we add it now, but aren't using it for anything, then we won't notice if we've done it wrong.  That can lead to problems that are _harder_ to fix later (like if we discover at some point in the future that we have 10 entries in the Fleet table, which we didn't notice because nothing was using it).

Reply:
Fair enough, removed in 206572d2f07bd714c7527e5fda07cce078c26bb7

I'm still inserting a record in `omicron.public.virtual_provisioning_collection` for the fleet, to track provisioning there.

Reply:
Deleting 230 lines with no loss of functionality speaks for itself 🙂